### PR TITLE
81503 System changes

### PR DESCRIPTION
**Pull request by bratli**
[AB#81503](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/81503)

NEED TO WAIT FOR WEBAPI BEFORE MERGING

NB!! need to discuss what to do with MCpkgs in db in prod! Leave it as is, create a sync job?
The problem I can think of is if someone want to edit an IPO in planned stage, and wants to add more MCpkgs outside the commpkg that current mc pkgs are on. Will not hit many users I think.

Add system to Mc pkgs in db.
Can pick comm pkg scope within system (before it was within subsystem)
Can pick mc pkg scope within system (before it was within commpkg)

Remove obscolete endpoint in ScopeController [AB#80829](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/80829)

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/CreateInvitation/CreateInvitationCommandHandler.cs [hunk 1]
```diff
@@ -386,8 +386,9 @@ private async Task AddCommPkgsAsync(Invitation invitation, IList<string> commPkg
             var initialCommPkg = commPkgDetailsList.FirstOrDefault();
             if (initialCommPkg != null)
             {
-                var initialSystem = initialCommPkg.System;
-                if (commPkgDetailsList.Any(commPkg => commPkg.System != initialSystem))
+                var initialSystem = initialCommPkg.System.Substring(0, initialCommPkg.System.LastIndexOf('|'));
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/InvitationCommands/CreateInvitation/CreateInvitationCommandHandler.cs:389]: make a get-property in CommPkg named SubSystem returning .System.LastIndexOf('|')
Do it for all places where you have this logic

Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/InvitationCommands/CreateInvitation/CreateInvitationCommandHandler.cs:389]: initialSubSystem ?

File: src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    public partial class AddSystemToMcPkgs : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "System",
+                table: "McPkgs",
+                type: "nvarchar(max)",
+                nullable: false,
+                defaultValue: "");
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs:14]: How will you migrate existing IPO's?

File: src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+namespace Equinor.ProCoSys.IPO.Infrastructure.Migrations
+{
+    public partial class AddSystemToMcPkgs : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "System",
+                table: "McPkgs",
+                type: "nvarchar(max)",
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs:12]: Is there no limits in any nvarchar columns in IPO?

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs [hunk 4]
```diff
@@ -37,6 +37,7 @@ public class InvitationTests
         private const string Title = "Title A";
         private const string Title2 = "Title B";
         private const string Description = "Description A";
+        private const string System = "12";
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs:40]: In real life, will it be possible to have McPkg under a System (not SubSystem)? Should McPkg constructor validate on a legal System-string (must have | and be 3 chars or more?)

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitations/GetInvitationsQueryHandlerTests.cs [hunk 5]
```diff
@@ -37,6 +37,7 @@ public class GetInvitationsQueryHandlerTests : ReadOnlyTestsBase
         const string _commPkgNo = "CommPkgNo";
         const string _commPkgNo2 = "CommPkgNo2";
         const string _mcPkgNo = "McPkgNo";
+        private const string _system = "12";
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitations/GetInvitationsQueryHandlerTests.cs:40]: use a legal subsystem-string?

**Review by janid1967 — CHANGES_REQUESTED** [review]
See comments

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/CreateInvitation/CreateInvitationCommandHandler.cs (hunk 1 above)
Comment (bratli) [reply src/Equinor.ProCoSys.IPO.Command/InvitationCommands/CreateInvitation/CreateInvitationCommandHandler.cs:389]: No, it should be System

File: src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs (hunk 2 above)
Comment (bratli) [reply src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs:14]: Need to discuss this with you and maybe Pål

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs (hunk 4 above)
Comment (bratli) [reply src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs:40]: No, this system is not valid, the system will always include a '|'. Will ad this check

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs:40]: The constructor should then throw Exception

Comment (bratli) [reply src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs:40]: Not ArgumentException ?

File: src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs (hunk 3 above)
Comment (bratli) [reply src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210302153400_AddSystemToMcPkgs.cs:12]: Yes there are, but have not added it here. Will add a max length of 40 to systems (do not see any systems longer than this in QP at least), and for mcPkg and commPkg nos.

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs (hunk 4 above)
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/InvitationTests.cs:40]: ArgumentException

**Review by janid1967 — APPROVED** [review]

**Pull request merged**